modules/context/repo.go

```diff
@@ -791,20 +791,45 @@ func RepoRefByType(refType RepoRefType) func(http.Handler) http.Handler {
 				if refType.RefTypeIncludesBranches() && ctx.Repo.GitRepo.IsBranchExist(refName) {
 					ctx.Repo.IsViewBranch = true
 
-					ctx.Repo.Commit, err = ctx.Repo.GitRepo.GetBranchCommit(refName)
-					if err != nil {
-						ctx.ServerError("GetBranchCommit", err)
-						return
+					path := strings.TrimLeft(ctx.Repo.TreePath, "/")
+					if len(path) > 0 {
+						ctx.Repo.Commit, err = ctx.Repo.GitRepo.GetCommitByRefAndPath(git.BranchPrefix+refName, path)
```

Comment:
`ctx.Repo.Commit` means currently browsing commit, not the treepath dir last commit

Reply:
Hmm, I think it's okay, Because when browsing a treepath, the last commit of this path is more usefull than last commit for current branch or tag. And it's same with gh.